tests/test_trainer.py

```diff
@@ -155,6 +157,17 @@ def test_plm(self):
 
 @require_torch
 class TrainerIntegrationTest(unittest.TestCase):
+    class SampleIterableDataset(IterableDataset):
```

Comment:
probably doesn't need to be nested either afaict.

Reply:
do might have to define it inside your test method. I think attributes declared above `__init__` are not protected by `@require_torch` :(

Reply:
or add an `if is_torch_available` outside the test class and define inside.